Clarity regarding dotnet Spark Job definition dependencies for Synapse

cutecycle opened the pull request:
There are a few things that were unintuitive about the documentation when we initially set up our .NET Spark project.
* We built using 2.1.0, and spent time trying to figure out why the DotNetRunner was timing out. 
* We found out we had to use 1.0.0 with our spark pool, and got a SerDe error when attempting to debug locally.

cutecycle committed:
Update spark-dotnet.md

cutecycle committed:
Update spark-dotnet.md

cutecycle committed:
Update spark-dotnet.md

cutecycle committed:
Update spark-dotnet.md

cutecycle committed:
Update spark-dotnet.md

JasonWHowell commented on articles/synapse-analytics/spark/spark-dotnet.md:
```diff
@@ -23,18 +23,42 @@ You can analyze data with .NET for Apache Spark through Spark batch job definiti
-1. Run the following commands to publish your app. Be sure to replace *mySparkApp* with the path to your app.
+1. Configure your dotnet application dependencies for compatibility with Synapse Spark.
+The required .NET Spark version will be noted in the Studio interface under your Spark Pool configuration. 
+![image](https://user-images.githubusercontent.com/12169942/161816341-ca13b48f-fb18-49bd-a61b-6af60bf9846e.png)
```
Images need to be hosted in docs, not on uploaded Github content.

JasonWHowell replied:
Image files go here: https://github.com/MicrosoftDocs/azure-docs/tree/main/articles/synapse-analytics/spark/media/<folder for this page>

cutecycle committed:
Add files via upload

cutecycle committed:
Update spark-dotnet.md

cutecycle committed:
Update spark-dotnet.md

ShannonLeavitt committed:
Update spark-dotnet.md

Fixing indentation (lines 29-47), adding alt text (line 29), lowercasing "console application" (line 32), capitalizing product name (line 31), adding period to end of sentence (line 111)

ShannonLeavitt committed:
Update spark-dotnet.md

Updating image coding to add gray border around image (line 29), increasing indentation (lines 32-47)
